### PR TITLE
Replace referencecs to Ratpack's @Nullable with jetbrains

### LINKED ISSUE #1274: Annotate source with JSR305 (Nullable/Nonnull)
This would be helpful to Kotlin consumers of Ratpack to get better tooling assistance for when values may be null.

https://github.com/ratpack/ratpack/blob/master/ratpack-base/src/main/java/ratpack/api/Nullable.java exists but there is not an equivalent for Nonnull. `TypeQualifierNickname` or something similar can be used if internal annotations types are desired.

### PR DESCRIPTION
**Pull request by michaelschlies**
fixes #1274

---
This change is [Reviewable](https://reviewable.io/reviews/ratpack/ratpack/1477)

### CODE REVIEW

**Comment by michaelschlies** [comment]
Replacing internal `@Nullable` could be breaking depending on if anyone is using reflection, this should be a 2.x change

File: ratpack-core/src/main/java/ratpack/file/FileSystemBinding.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 package ratpack.file;
 
 import com.google.common.reflect.TypeToken;
-import ratpack.api.Nullable;
```
Comment (johnrengelman) [inline ratpack-core/src/main/java/ratpack/file/FileSystemBinding.java]: @michaelschlies looks like your find/replace duplicated `;` throughout the code base.

Comment (michaelschlies) [reply ratpack-core/src/main/java/ratpack/file/FileSystemBinding.java]: Good catch @johnrengelman, I'll clean those up this afternoon

Comment (michaelschlies) [reply ratpack-core/src/main/java/ratpack/file/FileSystemBinding.java]: Ooops, finally got around to cleaning those up, CI running now

**Comment by johnrengelman** [comment]
@michaelschlies looks like there is one failing spec with this change: `ratpack.guice.GuiceRegistrySpec.can inject null using ratpack's Nullable annotation`

**Comment by johnrengelman** [comment]
I think you also need to rebase as there are some changes in master to fix test errors around some bad links.

**Comment by michaelschlies** [comment]
Ah, looking into a @Nullable conflict with Guice, brb with this one.

**Pull request closed without merging**